File: 058. Length of Last Word/58. Length of Last Word.cpp
//下面这种方法是第一种解法的优化版本，
//由于我们只关心最后一个单词的长度，
//所以开头有多少个空格其实我们并不在意，可以从字符串末尾开始，
//先将末尾的空格都去掉，然后开始找非空格的字符的长度即可。
class Solution {
public:
    int lengthOfLastWord(string s) {
        int right = s.size() - 1, res = 0;
        while (right >= 0 && s[right] == ' ') --right;
        while (right >= 0 && s[right] != ' ' ) {
            --right; 
            ++res;
        }
        return res;
    }
};

File: 058. Length of Last Word/58. Length of Last Word.java
/**
方法来源：https://blog.csdn.net/mikuluna/article/details/84347221
直接先去两边空格，然后倒叙，如果不是空格，则count++；
如果是空格，就直接退出循环。最后这个count就是单词长度

*/
class Solution {
    public int lengthOfLastWord(String s) {
        int len = s.length();
        s = s.trim();
        int count = 0;
        for(int i = s.length() - 1;i >= 0; i--){
            if(s.charAt(i) != ' '){
                count++;
            }else {
                break;
            }
        }
        return count;
    }
}